napari/layers/utils/text.py

```diff
@@ -39,193 +61,49 @@ class TextManager:
         The default value is 'translucent'
     visible : bool
         Set to true of the text should be displayed.
-
-    Attributes
-    ----------
-    text : array
-        the strings to be displayed
-    rotation : float
-        Angle of the text elements around the anchor point. Default value is 0.
-    anchor : str
-        The location of the text origin relative to the bounding box.
-        Should be 'center', 'upper_left', 'upper_right', 'lower_left', or 'lower_right'.
-    translation : np.ndarray
-        Offset from the anchor point.
-    color : array
-        Font color for the text
-    size : float
-        Font size of the text. Default value is 12.
-    The blending mode that determines how RGB and
-        alpha values of the layer visual get mixed. Allowed values are
-        {'opaque', 'translucent', and 'additive'}. Note that 'opaque` blending
-        is not recommended, as colors the bounding box surrounding the text.
-    visible : bool
-        Set to true of the text should be displayed.
     """
 
-    def __init__(
-        self,
-        text,
-        n_text,
-        properties={},
-        rotation=0,
-        translation=0,
-        anchor='center',
-        color='cyan',
-        size=12,
-        blending='translucent',
-        visible=True,
-    ):
-
-        self.events = EmitterGroup(
-            source=self,
-            auto_connect=True,
-            text=Event,
-            rotation=Event,
-            translation=Event,
-            anchor=Event,
-            color=Event,
-            size=Event,
-            blending=Event,
-            visible=Event,
-        )
-
-        self.events.block_all()
-        self._rotation = rotation
-        self._anchor = Anchor(anchor)
-        self._translation = translation
-        self._color = transform_color(color)[0]
-        self._size = size
-        self._blending = self._check_blending_mode(blending)
-        self._visible = visible
-
-        self._set_text(text, n_text, properties)
-        self.events.unblock_all()
-
-    @property
-    def values(self):
-        """np.ndarray: the text values to be displayed"""
-        return self._values
+    text: InitVar[Union[np.ndarray, str]]
+    n_text: InitVar[int]
+    properties: InitVar[Dict[str, np.ndarray]] = None
+
+    rotation: float = 0.0
+    translation: Annotated[np.ndarray, None, np.asarray] = np.asarray(0)
+    anchor: Annotated[Anchor, str, Anchor] = Anchor.CENTER
+    color: Annotated[
+        Union[np.ndarray, str], to_hex, transform_1_color
```

Comment:
see #1491... this is a good example of needing a better general representation of colors.  The difficulty in evaluating the equivalence of `np.ndarray` requires us to change the `__eq__` logic of the dataclass, (or coerce the value to something like hex when getting it, as done here).  It should be much easier to check for color equivalence.